src/EnergyPlus/HVACInterfaceManager.cc

```diff
@@ -383,12 +397,13 @@ void UpdateHVACInterface(EnergyPlusData &state,
 //***************
 
 // In-Place Right Shift by 1 of Array Elements
-void rshift1(Array1D<Real64> &a)
+void rshift1(std::array<Real64, DataConvergParams::ConvergLogStackDepth> &a)
 {
-    assert(a.size_bounded());
-    for (int i = a.u(), e = a.l(); i > e; --i) {
-        a(i) = a(i - 1);
+    Real64 lastVal = a[a.size() - 1];
+    for (unsigned int i = a.size() - 1; i > 0; --i) {
+        a[i] = a[i - 1];
     }
+    a[0] = lastVal;
```

Comment:
A custom rshift function was here for Array1D just for use in the following function, nowhere else in the code.  To keep this simple, I just altered it in place to operate on a std::array instead.  I would be open to using a standard library function but I didn't see an algorithm function for it in my first glance.

Reply:
You can use `memmove`, but for arrays this short, it's not worth it. You should probably declare this `inline`.

Reply:
Added inline to it, not sure if the compiler did anything with it, but it's there.

Reply:
You need to move it to the .hh file.